.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,135 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Binary'
+
+description: |-
+  Use this action to install and cache a binary for use within a GitHub workflow. This action wraps the functionality of go-getter to provide advanced donwload and checksum capabilities.
+
+inputs:
+  download_url:
+    description: 'The download URL for the binary. This must a valid go-getter URL. See https://github.com/hashicorp/go-getter#url-format'
+    required: true
+  checksum:
+    description: 'The checksum for the downloaded artifact. See https://github.com/hashicorp/go-getter#checksumming'
+    required: false
+  install_path:
+    description: 'The path on disk in which to install the binary (defaults to the current directory). This includes the folder and binary name (e.g. /path/to/my/binary). This action does not modify $PATH; if you want the binary to be accessible, make sure the parent folder is in $PATH.'
+    default: '.'
+    required: false
+  cache_key:
+    description: 'The cache key to use when caching the downloaded binary. If this value is empty, caching will be disabled.'
+    required: false
+  add_to_path:
+    description: 'Add the install_path to the $PATH variable.'
+    default: 'false'
+    required: false
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Validate inputs'
+      shell: 'bash'
+      env:
+        DOWNLOAD_URL: '${{ inputs.download_url }}'
+      run: |
+        # Validate inputs
+        if [[ -z "${DOWNLOAD_URL}" ]]; then
+            echo "::error ::download_url is a required input"
+            exit 1
+        fi
+
+    - name: 'Restore cache binary'
+      id: 'cache-binary'
+      if: |
+        inputs.cache_key != ''
+      uses: 'actions/cache/restore@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8' # ratchet:actions/cache/restore@v3
+      with:
+        path: '${{ inputs.install_path }}'
+        key: 'setup-binary-${{ inputs.cache_key }}'
+
+    - name: 'Install hashicorp/go-getter'
+      if: |
+        steps.cache-guardian.outputs.cache-hit != 'true'
+      shell: 'bash'
+      env:
+        VERSION: '1.7.1'
```

Comment:
Two things:

1. I'd suggest naming this "GOGETTER_VERSION" or something more specific.
2. Should this be an input controlled by the user? (I'm leaning "no", but wanted your thoughts).